spec/lib/extensions/database_configuration_spec.rb

```diff
@@ -11,14 +13,20 @@
 
   context "ERB in the template" do
     before do
-      database_config = StringIO.new "---\nvalue: <%= 1 %>\n"
-      allow(database_config).to receive(:exist?).and_return(true)
-      allow(Pathname).to receive(:new).and_return(database_config)
+      @database_config = Tempfile.new
+      @database_config.write("---\nvalue: <%= 1 %>\n")
+      @database_config.rewind
```

Comment:
This is not necessary as the method under test will use a separate file object

Reply:
Bombs without it.

Reply:
I think that's likely because `write` doesn't flush to the filesystem immediately. You could probably use a `sync` call here, but I think it would be better to use [`Tempfile.create`](https://ruby-doc.org/stdlib-2.4.0/libdoc/tempfile/rdoc/Tempfile.html#method-c-create) to grab a path then `File.write` which will sync for you. Then in the after (or second part of the `around`) you will delete the file path rather than closing the file.